Makefile

```diff
@@ -251,6 +285,17 @@ ${BUILD_DIR}/%/bin/pebble: phony_explicit
 # build for pebble
 	$(run_go_build)
 
+${SIMPLESTREAMS_DIR}/tools/released/juju-${JUJU_VERSION}-%.tgz: phony_explicit juju $(BUILD_AGENT_TARGETS)
+	@echo "Packaging simplestream tools for juju ${JUJU_VERSION} on $*"
+	@mkdir -p ${SIMPLESTREAMS_DIR}/tools/released
+	@tar czf "$@" -C $(call bin_platform_path,$(subst -,/,$*)) .
+
+.PHONY: simplestreams
+simplestreams: juju juju-metadata ${SIMPLESTREAMS_TARGETS}
+	@juju metadata generate-agents -d ${SIMPLESTREAMS_DIR} --clean --prevent-fallback ;
+	@echo "\nRun export JUJU_METADATA_SOURCE=\"${SIMPLESTREAMS_DIR}\" "
```

Comment:
If `JUJU_METADATA_SOURCE` is set we should just use that directly instead of `${BUILD_DIR}/simplestreams`
And we can then also omit the "Run export ...." output

Reply:
I have made some changes so the Makefile now runs off of JUJU_METADATA_SOURCE if it is defined. However doing that kind of conditional logic in recipe's is extremely hard and not what Make does well. I have extended the echo to make it more conditional in the wording.